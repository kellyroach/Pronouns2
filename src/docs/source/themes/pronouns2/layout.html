{#
    basic/layout.html
    ~~~~~~~~~~~~~~~~~

    Master layout template for Sphinx themes.

    :copyright: Copyright 2007-2016 by the Sphinx team, see AUTHORS.
    :license: BSD, see LICENSE for details.
#}
{%- block doctype -%}
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
{%- endblock %}
{%- set reldelim1 = reldelim1 is not defined and ' &gt;' or reldelim1 %}
{%- set render_sidebar = (not embedded) and (not theme_nosidebar|tobool) and
                         (sidebars != []) %}
{%- if theme_collapsiblesidebar|tobool %}
  {%- set script_files = script_files + ['_static/sidebar.js'] %}
{%- endif %}
{%- set url_root = pathto('', 1) %}
{# XXX necessary? #}
{%- if url_root == '#' %}{% set url_root = '' %}{% endif %}
{%- if not embedded and docstitle %}
  {%- set titlesuffix = " &#8212; "|safe + docstitle|e %}
{%- else %}
  {%- set titlesuffix = "" %}
{%- endif %}

{%- macro relbar() %}
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>{{ _('Navigation') }}</h3>
      <ul>
        {%- block rootrellink %}
        <li class="nav-item nav-item-0"><a href="{{ pathto(master_doc) }}">Pronouns2</a>{{ reldelim1 }}</li>
        {%- endblock %}
        {%- for parent in parents %}
          <li class="nav-item nav-item-{{ loop.index }}"><a href="{{ parent.link|e }}" {% if loop.last %}{{ accesskey("U") }}{% endif %}>{{ parent.title }}</a>{{ reldelim1 }}</li>
        {%- endfor %}
        {%- block relbaritems %} {% endblock %}
      </ul>
    </div>
{%- endmacro %}

{%- macro sidebar() %}
      {%- if render_sidebar %}
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
          {%- block sidebarlogo %}
          {%- if logo %}
            <p class="logo"><a href="{{ pathto(master_doc) }}">
              <img class="logo" src="{{ pathto('_static/' + logo, 1) }}" alt="Logo"/>
            </a></p>
          {%- endif %}
          {%- endblock %}
          {%- if sidebars != None %}
            {#- new style sidebar: explicitly include/exclude templates #}
            {%- for sidebartemplate in sidebars %}
            {%- include sidebartemplate %}
            {%- endfor %}
          {%- else %}
            {#- old style sidebars: using blocks -- should be deprecated #}
            {%- block sidebartoc %}
            {%- include "localtoc.html" %}
            {%- endblock %}
            {%- block sidebarrel %}
            {%- include "relations.html" %}
            {%- endblock %}
            {%- if customsidebar %}
            {%- include customsidebar %}
            {%- endif %}
          {%- endif %}
        </div>
      </div>
      {%- endif %}
{%- endmacro %}

{%- macro script() %}
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '{{ url_root }}',
        VERSION:     '{{ release|e }}',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '{{ '' if no_search_suffix else file_suffix }}',
        HAS_SOURCE:  {{ has_source|lower }},
        SOURCELINK_SUFFIX: '{{ sourcelink_suffix }}'
      };
    </script>
    {%- for scriptfile in script_files %}
    <script type="text/javascript" src="{{ pathto(scriptfile, 1) }}"></script>
    {%- endfor %}
{%- endmacro %}

{%- macro css() %}
    {%- for css in css_files %}
      {%- if css|attr("filename") %}
    {{ css_tag(css) }}
      {%- else %}
    <link rel="stylesheet" href="{{ pathto(css, 1)|e }}" type="text/css" />
      {%- endif %}
    {%- endfor %}
{%- endmacro %}

<html xmlns="http://www.w3.org/1999/xhtml"{% if language is not none %} lang="{{ language }}"{% endif %}>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset={{ encoding }}" />
    {{ metatags }}
    {%- block htmltitle %}
    <title>{{ title|striptags|e }}{{ titlesuffix }}</title>
    {%- endblock %}
    {{ css() }}
    {%- if not embedded %}
    {{ script() }}
    {%- if use_opensearch %}
    <link rel="search" type="application/opensearchdescription+xml"
          title="{% trans docstitle=docstitle|e %}Search within {{ docstitle }}{% endtrans %}"
          href="{{ pathto('_static/opensearch.xml', 1) }}"/>
    {%- endif %}
    {%- if favicon %}
    <link rel="shortcut icon" href="{{ pathto('_static/' + favicon, 1) }}"/>
    {%- endif %}
    {%- endif %}
{%- block linktags %}
    {%- if hasdoc('about') %}
    <link rel="author" title="{{ _('About these documents') }}" href="{{ pathto('about') }}" />
    {%- endif %}
    {%- if hasdoc('genindex') %}
    <link rel="index" title="{{ _('Index') }}" href="{{ pathto('genindex') }}" />
    {%- endif %}
    {%- if hasdoc('search') %}
    <link rel="search" title="{{ _('Search') }}" href="{{ pathto('search') }}" />
    {%- endif %}
    {%- if hasdoc('copyright') %}
    <link rel="copyright" title="{{ _('Copyright') }}" href="{{ pathto('copyright') }}" />
    {%- endif %}
    {%- if next %}
    <link rel="next" title="{{ next.title|striptags|e }}" href="{{ next.link|e }}" />
    {%- endif %}
    {%- if prev %}
    <link rel="prev" title="{{ prev.title|striptags|e }}" href="{{ prev.link|e }}" />
    {%- endif %}
{%- endblock %}
{%- block extrahead %} {% endblock %}
  </head>
  <body role="document" onload="MM_preloadImages('{{ pathto('_static/Images/arrow01c.gif', 1) }}">
{%- block header %}{% endblock %}
    <form method="get" action="{{ pathto('search') }}" style="margin:0px">
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
      <table width="100%" border="0" cellspacing="0" cellpadding="4">
        <tr>
          <td rowspan="2" bgcolor="#539933"><img src="{{ pathto('_static/Images/PlanetQuantumWhite50.png', 1) }}" width="50" height="50"/></td>
          <td colspan="4" bgcolor="#CCEECC"><table width="100%" border="0" cellspacing="0" cellpadding="0">
              <tr>
                <td width="25%" align="center" bgcolor="#99CC99">&nbsp;<a href="{{ pathto('Index') }}">Home</a>&nbsp;</td>
                <td align="right" width="74%"><input name="q" type="text" title="Enter words to search for and click 'Search'"
                    style="width:250px"/></td>
                <td align="center" width="1%"></td>
                <td align="center" width="0%"><input type="submit" value="Search"/></td>
              </tr>
            </table></td>
{%- block up %}
  {%- set up = pathto(master_doc) %}
  {%- if parents|length > 0 %}
  {%- set parent = parents[ parents|length - 1 ] %}
  {%- set up = parent.link|e %}
  {%- endif %}
          <td rowspan="2" width="30" bgcolor="#539933"><a
          href="{{ up }}"
          onmouseout="MM_swapImgRestore()"
          onmouseover="MM_swapImage('Up','','{{ pathto('_static/Images/arrow01c.gif', 1) }}',1)"><img
          src="{{ pathto('_static/Images/arrow02c.gif', 1) }}" alt="Up" name="Up" width="30" height="40" border="0"/></a></td>
        </tr>
{%- endblock %}
        <tr>
          <td width="40" bgcolor="#539933">{%- if prev %}<a
              href="{{ prev.link|e }}"
              onmouseout="MM_swapImgRestore()"
              onmouseover="MM_swapImage('Previous','','{{ pathto('_static/Images/arrow01d.gif', 1) }}',1)"><img
                  src="{{ pathto('_static/Images/arrow02d.gif', 1) }}"
                  alt="Previous"
                  name="Previous"
                  width="40"
                  height="30"
                  border="0"/></a>{%- endif %}</td>
          <td width="100%" bgcolor="#539933"><font color="#FFFFFF">{{ relbar() }}</font></td>
          <td width="40" bgcolor="#539933">{%- if next %}<a
              href="{{ next.link|e }}"
              onmouseout="MM_swapImgRestore()"
              onmouseover="MM_swapImage('Next','','{{ pathto('_static/Images/arrow01e.gif', 1) }}',1)"><img
                  src="{{ pathto('_static/Images/arrow02e.gif', 1) }}"
                  alt="Next"
                  name="Next"
                  width="40"
                  height="30"
                  border="0"/></a>{%- endif %}</td>
        </tr>
      </table>
    </form>

{%- block content %}
  {%- block sidebar1 %} {# possible location for sidebar #} {% endblock %}

    <div class="document">
  {%- block document %}
      <div class="documentwrapper">
      {%- if render_sidebar %}
        <div class="bodywrapper">
      {%- endif %}
          <div class="body" role="main">
            {% block body %} {% endblock %}
          </div>
      {%- if render_sidebar %}
        </div>
      {%- endif %}
      </div>
  {%- endblock %}

  {%- block sidebar2 %}{{ sidebar() }}{% endblock %}
      <div class="clearer"></div>
    </div>
{%- endblock %}
    <table width="100%" border="0" cellspacing="0" cellpadding="2" bgcolor="#CCEECC">
      <tr>
        <td width="50%">&#169;2024 Kelly Roach</td>
        <td width="40%" align="right"><a id="email" href="mailto:kellybrianroach@outlook.com">Kelly Roach</a></td>
      </tr>
    </table>
  </body>
</html>
