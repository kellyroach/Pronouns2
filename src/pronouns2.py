################################################################
#
#    pronouns2.py
#
################################################################

"""Main project Python module to import which imports everything else."""

import argparse
from doc import *
from find_files import *
from varying_rho import *

################################################################
#
#     Module Dependencies
#
# pronouns2
#   find_files
#   varying_rho
#     doc
#       demo
#         examples
#         table_interp
#           view
#             lexicon
#               node_proc
#                 trc
#                   globals
#         table_proc
#           abstract
#             modern
#               tex
#               view
#           chaining
#             modern
#             spread
#               trc
#           latex
#             modern
#           parse_proc
#             lexicon
#           secondary_uty
#             primary_uty
#               trc
#             modern
#             lexicon
#           txt
#             view
#
# NOTE: Except for trc.py, any module which must depend on
# globals.py should indirectly depend on globals.py through
# dependence on some other Python module or trc.py as the
# last resort.  Our strict policy keeps the tree simpler.
################################################################

def main():
    """Main entry point for the pronouns2.py command-line tool.
    
    Supports command line formats:
    python pronouns2.py
    python pronouns2.py "10.1"
    python pronouns2.py --file="MyFile"
    python pronouns2.py "10.1" --file="MyFile"
    python pronouns2.py --help
    
    Additional boolean options from Manager.state can be specified:
    --option_name          (sets option to True)
    --option_name=True/False
    
    Special options:
    --file_type=.txt/.tex (sets output file type)
    --chaining_rho=0.0-1.0 (sets chaining parameter)
    
    Returns:
        None
    """

    # Create parser
    parser = argparse.ArgumentParser(description='Generate analyses for examples')
    # Add positional argument for numbers (optional)
    parser.add_argument('numbers', nargs='?', default=None,
                       help='Example numbers to document (e.g. "10.1")')
    # Add --file argument
    parser.add_argument('--file', default="../docs/Demo",
                       help='Output file path (default: ../docs/Demo)')
    # Add arguments for all Manager.state keys (except 'stream')
    valid_state_keys = set(Manager.manager.state.keys()) - {'stream'}
    for key in valid_state_keys:
        if key == 'chaining_rho':
            parser.add_argument(f'--{key}', type=float,
                              help='Chaining parameter (0.0-1.0)')
        elif key == 'file_type':
            parser.add_argument(f'--{key}', choices=['.txt', '.tex'],
                              help='Output file type')
        else:
            parser.add_argument(f'--{key}', nargs='?', const=True, type=bool,
                              help=f'Set {key} (default: {Manager.manager.state[key]})')
    # Parse arguments
    args = parser.parse_args()
    # Build kwargs dictionary for doc()
    kwargs = {}
    # Handle special cases first
    if args.file_type:
        kwargs['file_type'] = FileType.from_ext(args.file_type)
    if args.chaining_rho is not None:
        if not 0.0 <= args.chaining_rho <= 1.0:
            parser.error("--chaining_rho must be between 0.0 and 1.0")
        kwargs['chaining_rho'] = args.chaining_rho
    # Add all other boolean options from valid_state_keys
    for key in valid_state_keys:
        if key not in {'chaining_rho', 'file_type'}:
            value = getattr(args, key)
            if value is not None:
                kwargs[key] = value
    # Call doc() with parsed arguments
    full_path = doc(numbers=args.numbers, file=args.file, **kwargs)
    print(f"Created: {full_path}")

if __name__ == '__main__':
    main()
