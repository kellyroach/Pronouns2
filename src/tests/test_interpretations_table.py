################################################################
#
#    test_interpretations_table.py
#
################################################################

from unittest import TestCase
from tests.utils.run_test_doc import *
from pronouns2 import *

def run_test_interpretations_table(file: str, **kwargs) -> None:
    """Shared subroutine for testing demo_write_interpretations_table.

    Args:
        file (str): File name.
    """
    kwargs.update({'info': True,
                   'features_table': False,
                   'abstract_diagram': False,
                   'nodes_after_chaining': False,
                   'nodes_table': False,
                   'chaining_diagram': False,
                   'chaining_table': False,
                   'interpretations_table': True,
                   'summary_table': False,
                   'lexicon_table': False,
                   'debug': False,
                   'trace': False})
    run_test_doc(file=file, **kwargs)

class TestInterpretationsTable(TestCase):
    def test_interpretations_table_txt(self):
        """Test interpretations_table for file_type=FileType.TXT."""
        run_test_interpretations_table(file="test_interpretations_table.txt", file_type=FileType.TXT)
    def test_interpretations_table_tex(self):
        """Test interpretations_table for file_type=FileType.TEX."""
        run_test_interpretations_table(file="test_interpretations_table.tex", file_type=FileType.TEX)

