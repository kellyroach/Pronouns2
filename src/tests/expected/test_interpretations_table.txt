################################################################
#
#     test_interpretations_table.txt
#
################################################################

################################################################
#
#     (1.1) The boy who was fooling her kissed the girl who loved him.
#
################################################################

______INTERPRETATIONS______
!                         !
! girlB^herA    boyB^himA !
!_________________________!

################################################################
#
#     (1.2) *John killed herself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.4) Some students think they are smarter than they are.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.6) My uncle has never ridden a camel but his brother has, although it was lame.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.10) I like the fresh candy better than the stale PHI.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.11) If John can, he will do it.
#
################################################################

INTERPRETATIONS
!           !
! JohnB^heA !
!___________!

################################################################
#
#     (1.12) If he can, John will do it.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.13) John will do it if he can.
#
################################################################

INTERPRETATIONS
!           !
! JohnB^heA !
!___________!

################################################################
#
#     (1.14) He will do it if John can.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.24) I have a cat at home, but hate it.
#
################################################################

_____INTERPRETATIONS______
!                        !
! catB^PHIA    homeB^itA !
!  IB^PHIA    homeB^itA  !
!________________________!

################################################################
#
#     (1.25) I want to get a cat for myself.
#
################################################################

__INTERPRETATIONS__
!                 !
! IC^PHIB^myselfA !
!_________________!

################################################################
#
#     (1.27) The men took off their hats.
#
################################################################

INTERPRETATIONS
!             !
! menB^theirA !
!_____________!

################################################################
#
#     (1.51) The man who lives next door said that he would mow my lawn.
#
################################################################

INTERPRETATIONS
!          !
! manB^heA !
!__________!

################################################################
#
#     (1.52) Somebody seduced Bill's sister, but no one will ever seduce Jack’s and she knows it.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.66) Realizing that he was unpopular didn't disturb Oscar.
#
################################################################

__INTERPRETATIONS__
!                 !
! OscarD^PHIB^heA !
!_________________!

################################################################
#
#     (1.70) My neighbor who is pregnant said that she was very happy.
#
################################################################

_INTERPRETATIONS__
!                !
! neighborB^sheA !
!________________!

################################################################
#
#     (1.74) The pilot who shot at it hit the Mig that chased him.
#
################################################################

______INTERPRETATIONS______
!                         !
! MigB^itA    pilotB^himA !
!_________________________!

################################################################
#
#     (1.86) The mosquito which bit Algernon was killed by him.
#
################################################################

_INTERPRETATIONS__
!                !
! AlgernonB^himA !
!________________!

################################################################
#
#     (1.87) The mosquito which bit him was killed by Algernon.
#
################################################################

_INTERPRETATIONS__
!                !
! mosquitoB^himA !
! AlgernonB^himA !
!________________!

################################################################
#
#     (1.88) Algernon killed the mosquito which bit him.
#
################################################################

_INTERPRETATIONS__
!                !
! AlgernonB^himA !
! mosquitoB^himA !
!________________!

################################################################
#
#     (1.89) He killed the mosquito which bit Algernon.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.91) After John Adams woke up, he was hungry.
#
################################################################

INTERPRETATIONS
!           !
! JohnB^heA !
!___________!

################################################################
#
#     (1.92) That Oscar was unpopular didn't disturb him.
#
################################################################

INTERPRETATIONS
!             !
! OscarB^himA !
!_____________!

################################################################
#
#     (1.93) For your brother to refuse to pay taxes would get him into trouble.
#
################################################################

___INTERPRETATIONS____
!                    !
! brotherD^PHIB^himA !
!____________________!

################################################################
#
#     (1.94) Anna's complaining about Peter infuriated him.
#
################################################################

INTERPRETATIONS
!             !
! PeterB^himA !
!_____________!

################################################################
#
#     (1.95) The possibility that Fred will be unpopular doesn’t bother him.
#
################################################################

INTERPRETATIONS
!            !
! FredB^himA !
!____________!

################################################################
#
#     (1.96) After he woke up, John Adams was hungry.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.97) That he was unpopular didn't disturb Oscar.
#
################################################################

INTERPRETATIONS
!            !
! OscarB^heA !
!____________!

################################################################
#
#     (1.98) For him to refuse to pay taxes would get your brother into trouble.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.99) Anna's complaining about him infuriated Peter.
#
################################################################

INTERPRETATIONS
!             !
! PeterB^himA !
!_____________!

################################################################
#
#     (1.100) The possibility that he will be unpopular doesn’t bother Fred.
#
################################################################

INTERPRETATIONS
!           !
! FredB^heA !
!___________!

################################################################
#
#     (1.101) John Adams was hungry after he woke up.
#
################################################################

INTERPRETATIONS
!           !
! JohnB^heA !
!___________!

################################################################
#
#     (1.102) Oscar wasn't disturbed that he was unpopular.
#
################################################################

INTERPRETATIONS
!            !
! OscarB^heA !
!____________!

################################################################
#
#     (1.103) It would get your brother into trouble for him to refuse to pay taxes.
#
################################################################

_________INTERPRETATIONS__________
!                                !
! brotherC^himA    troubleB^PHIA !
!       brotherD^himB^PHIA       !
!________________________________!

################################################################
#
#     (1.104) Peter was infuriated at Anna's complaining about him.
#
################################################################

INTERPRETATIONS
!             !
! PeterB^himA !
!_____________!

################################################################
#
#     (1.105) Fred isn't bothered by the possibility that he will be unpopular.
#
################################################################

INTERPRETATIONS
!           !
! FredB^heA !
!___________!

################################################################
#
#     (1.106) *He was hungry after John Adams woke up.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.107) *He wasn't disturbed that Oscar was unpopular.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.108) *It would get him into trouble for your brother to refuse to pay taxes.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.109) *He was infuriated at Anna's complaining about Peter.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.110) *He isn't bothered by the possibility that Fred will be unpopular.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.111) Penelope cursed Peter and slandered him.
#
################################################################

_________INTERPRETATIONS_________
!                               !
! PenelopeB^PHIA    PeterB^himA !
!_______________________________!

################################################################
#
#     (1.112) *Penelope cursed him and slandered Peter.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (1.113) The interest in visiting Las Vegas that Mary displayed is typical of gamblers.
#
################################################################

____________INTERPRETATIONS____________
!                                     !
! interestC^PHI1A    Las VegasB^PHI2A !
!  interestC^PHI1A    gamblersB^PHI2A !
!        interestD^PHI1B^PHI2A        !
!    MaryB^PHI1A    interestB^PHI2A   !
!   MaryB^PHI1A    Las VegasB^PHI2A   !
!    MaryB^PHI1A    gamblersB^PHI2A   !
!          MaryC^PHI1B^PHI2A          !
!  gamblersC^PHI1A    interestB^PHI2A !
! gamblersC^PHI1A    Las VegasB^PHI2A !
!        gamblersD^PHI1B^PHI2A        !
!_____________________________________!

################################################################
#
#     (5.2) June hates flowers, but she waters them anyway.
#
################################################################

________INTERPRETATIONS_________
!                              !
! JuneB^sheA    flowersB^themA !
!______________________________!

################################################################
#
#     (8.1) John wants to give June a present, but he isn't sure she’ll like it.
#
################################################################

________________INTERPRETATIONS_________________
!                                              !
! JohnD^PHID^heA    JuneB^sheA    presentB^itA !
!______________________________________________!

################################################################
#
#     (10.1) John wants to give June a present, but he isn't sure she’ll like it.
#
################################################################

________________INTERPRETATIONS_________________
!                                              !
! JohnD^PHID^heA    JuneB^sheA    presentB^itA !
!______________________________________________!

################################################################
#
#     (10.2) Janet saw herself.
#
################################################################

__INTERPRETATIONS__
!                 !
! JanetB^herselfA !
!_________________!

################################################################
#
#     (10.3) Janet saw her.
#
################################################################

INTERPRETATIONS
!             !
! JanetB^herA !
!_____________!

################################################################
#
#     (10.4) *Janet saw himself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (10.5) The men threw a smokescreen around themselves.
#
################################################################

__INTERPRETATIONS___
!                  !
! menB^themselvesA !
!__________________!

################################################################
#
#     (10.6) The men found a smokescreen around them.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (10.7) The men found a smokescreen to be around them.
#
################################################################

INTERPRETATIONS
!            !
! menB^themA !
!____________!

################################################################
#
#     (10.8) The men found a smokescreen and it was around them.
#
################################################################

_________INTERPRETATIONS__________
!                                !
! smokescreenB^itA    menB^themA !
!________________________________!

################################################################
#
#     (10.9) I told John to protect himself.
#
################################################################

____INTERPRETATIONS____
!                     !
! JohnC^PHIB^himselfA !
!_____________________!

################################################################
#
#     (10.10) I told John to protect me.
#
################################################################

____INTERPRETATIONS_____
!                      !
!      JohnB^PHIA      !
! IB^meA    JohnB^PHIA !
!       IC^PHIA        !
!______________________!

################################################################
#
#     (10.11) I told John to protect myself.
#
################################################################

__INTERPRETATIONS__
!                 !
! IC^PHIB^myselfA !
!_________________!

################################################################
#
#     (11.12) Jack's house burned down, but he rebuilt it.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (11.26) John owns some sheep and Harry vaccinates them.
#
################################################################

INTERPRETATIONS_
!              !
! sheepB^themA !
!______________!

################################################################
#
#     (11.27) Mary danced with many boys and they found her interesting.
#
################################################################

_______INTERPRETATIONS_______
!                           !
! boysB^theyA    MaryB^herA !
!___________________________!

################################################################
#
#     (11.28) John lost a pen yesterday and Bill found one today.
#
################################################################

INTERPRETATIONS
!            !
! JohnB^oneA !
! penB^oneA  !
!____________!

################################################################
#
#     (11.29) John claimed to have found the solution to the problem, but Bill was sure he had found it.
#
################################################################

______________INTERPRETATIONS_______________
!                                          !
! JohnC^PHIA    BillB^heA    solutionB^itA !
! JohnC^PHIA    BillB^heA    problemB^itA  !
!     JohnD^PHIC^heA    solutionB^itA      !
!      JohnD^PHIC^heA    problemB^itA      !
!__________________________________________!

################################################################
#
#     (11.30) John wants to catch a fish and eat it for supper.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (11.31) No one would put the blame on himself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (11.32) Sue told Sandy about herself.
#
################################################################

__INTERPRETATIONS__
!                 !
!  SueB^herselfA  !
! SandyB^herselfA !
!_________________!

################################################################
#
#     (11.33) *Jill kept talking about himself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (11.34) Does Jack's making a pig of himself bother Bill?
#
################################################################

_INTERPRETATIONS_
!               !
! pigB^himselfA !
!_______________!

################################################################
#
#     (11.35) John wants to give June a present, but he is afraid she won’t like it.
#
################################################################

________________INTERPRETATIONS_________________
!                                              !
! JohnD^PHID^heA    JuneB^sheA    presentB^itA !
!______________________________________________!

################################################################
#
#     (11.36) Ernie doesn't like Bernie, because he is such an asshole.
#
################################################################

INTERPRETATIONS
!             !
!  ErnieB^heA !
! BernieB^heA !
!_____________!

################################################################
#
#     (12.2) Mary's father killed himself.
#
################################################################

__INTERPRETATIONS___
!                  !
! fatherB^himselfA !
!__________________!

################################################################
#
#     (12.3) *Mary's father killed him.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (12.4) *Mary's father killed herself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (12.5) Mary's father killed her.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (12.6) The father of Mary killed himself.
#
################################################################

__INTERPRETATIONS___
!                  !
! fatherB^himselfA !
!__________________!

################################################################
#
#     (12.7) *The father of Mary killed him.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (12.8) *The father of Mary killed herself.
#
################################################################

_INTERPRETATIONS__
!                !
! MaryB^herselfA !
!________________!

################################################################
#
#     (12.9) The father of Mary killed her.
#
################################################################

INTERPRETATIONS
!            !
! MaryB^herA !
!____________!

################################################################
#
#     (12.11) Mary's mother cooks only for herself.
#
################################################################

__INTERPRETATIONS___
!                  !
! motherB^herselfA !
!__________________!

################################################################
#
#     (12.12) Mary's mother cooks only for her.
#
################################################################

INTERPRETATIONS_
!              !
! motherB^herA !
!______________!

################################################################
#
#     (12.13) Mary's mother cooks only for her mother.
#
################################################################

_INTERPRETATIONS_
!               !
! mother1B^herA !
!_______________!

################################################################
#
#     (13.2) It was difficult to sketch myself.
#
################################################################

__INTERPRETATIONS___
!                  !
! I0C^PHIB^myselfA !
!__________________!

################################################################
#
#     (13.5a) It was difficult for me to sketch myself.
#
################################################################

__INTERPRETATIONS__
!                 !
!   meB^myselfA   !
! I0C^meB^myselfA !
!_________________!

################################################################
#
#     (13.6a) It was difficult for you to sketch yourself.
#
################################################################

____INTERPRETATIONS_____
!                      !
!    youB^yourselfA    !
! you0C^youB^yourselfA !
!______________________!

################################################################
#
#     (13.7a) It was difficult for him to sketch himself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.8a) It was difficult for her to sketch herself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.9a) It was difficult for us to sketch ourselves.
#
################################################################

_INTERPRETATIONS__
!                !
! usB^ourselvesA !
!________________!

################################################################
#
#     (13.10a) It was difficult for you to sketch yourselves.
#
################################################################

_____INTERPRETATIONS______
!                        !
!    youB^yourselvesA    !
! you0C^youB^yourselvesA !
!________________________!

################################################################
#
#     (13.11a) It was difficult for them to sketch themselves.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.5b) It was difficult to sketch myself.
#
################################################################

__INTERPRETATIONS___
!                  !
! I0C^PHIB^myselfA !
!__________________!

################################################################
#
#     (13.6b) It was difficult to sketch yourself.
#
################################################################

____INTERPRETATIONS_____
!                      !
! you0C^PHIB^yourselfA !
!______________________!

################################################################
#
#     (13.7b) *It was difficult to sketch himself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.8b) *It was difficult to sketch herself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.9b) It was difficult to sketch ourselves.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.10b) It was difficult to sketch yourselves.
#
################################################################

_____INTERPRETATIONS______
!                        !
! you0C^PHIB^yourselvesA !
!________________________!

################################################################
#
#     (13.11b) *It was difficult to sketch themselves.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.12a) Nurse Bob Breezy gave up drawing.
#
################################################################

INTERPRETATIONS
!  !
!  !
!__!

################################################################
#
#     (13.12b) [Bob] It was difficult to sketch himself.
#
################################################################

___INTERPRETATIONS____
!                    !
! BobC^PHIB^himselfA !
!____________________!

################################################################
#
#     (13.13a) Astronaut Linda Smith gave up drawing.
#
################################################################

INTERPRETATIONS
!  !
!  !
!__!

################################################################
#
#     (13.13b) [Linda] It was difficult to sketch herself.
#
################################################################

____INTERPRETATIONS_____
!                      !
! LindaC^PHIB^herselfA !
!______________________!

################################################################
#
#     (13.14a) The bank embezzlers gave up drawing.
#
################################################################

INTERPRETATIONS
!  !
!  !
!__!

################################################################
#
#     (13.14b) [the bank embezzlers] It was difficult to sketch themselves.
#
################################################################

________INTERPRETATIONS_________
!                              !
! embezzlersC^PHIB^themselvesA !
!______________________________!

################################################################
#
#     (13.15) *It was difficult to sketch himself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.16) *It was difficult to sketch herself.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.17) *It was difficult to sketch themselves.
#
################################################################

INTERPRETATIONS
!      !
! NONE !
!______!

################################################################
#
#     (13.18) [Bob] It was difficult to sketch himself.
#
################################################################

___INTERPRETATIONS____
!                    !
! BobC^PHIB^himselfA !
!____________________!

################################################################
#
#     (13.19) [Linda] It was difficult to sketch herself.
#
################################################################

____INTERPRETATIONS_____
!                      !
! LindaC^PHIB^herselfA !
!______________________!

################################################################
#
#     (13.20) [the bank embezzlers] It was difficult to sketch themselves.
#
################################################################

________INTERPRETATIONS_________
!                              !
! embezzlersC^PHIB^themselvesA !
!______________________________!

################################################################
#
#     (13.21) It was difficult to sketch myself.
#
################################################################

__INTERPRETATIONS___
!                  !
! I0C^PHIB^myselfA !
!__________________!

################################################################
#
#     (13.22) [toy] Give me that!
#
################################################################

____________INTERPRETATIONS____________
!                                     !
!       you0B^PHIA    toyB^thatA      !
! I0B^meA    you0B^PHIA    toyB^thatA !
!        I0C^PHIA    toyB^thatA       !
!_____________________________________!
